File: iteration2/src/course/ElectiveCourse.java
package iteration2.src.course;


import iteration2.src.human.Assistant;
import iteration2.src.human.Grade;
import iteration2.src.human.Lecturer;
import iteration2.src.human.Student;

import java.util.List;

public class ElectiveCourse extends Course {

    public ElectiveCourse(String code, String name, int credits, int theoreticalHours, int appliedHours,
                          Grade firstYearToTake, Season firstSeasonToTake, List<Lecturer> lecturers, List<Assistant> assistants){
        super(code,name,credits,theoreticalHours,appliedHours,firstYearToTake,firstSeasonToTake,lecturers,assistants);
    }

    @Override
    public void addPrerequisite(Course prerequisite){
        return;
    }

    public boolean isMaxChoosableNumberExceeded(Student student, Season season){

       return false;

    }
}

File: iteration3/ElectiveCourse.py
import abc
from typing import List

from iteration3.Course import Course
from iteration3.Assistant import Assistant
from iteration3.Grade import Grade
from iteration3.Lecturer import Lecturer
from iteration3.Student import Student

class ElectiveCourse(Course):
    def __init__(self, code, name, credits, theoretical_hours, applied_hours, first_year_to_take, first_season_to_take, lecturers, assistants):
        super().__init__(code, name, credits, theoretical_hours, applied_hours, first_year_to_take, first_season_to_take, lecturers, assistants)

    @abc.abstractmethod
    def add_prerequisite(self, prerequisite):
        pass
